Implement tab-completion of device names for ./flash.sh

dHutchings opened the pull request:
Tab completion is setup on a linux machine as part of running ./setup.sh

Only works on linux. However, since the setup for compilation of hibike
also only works on linux, I think this is fine.

Done based on this documenation: https://debian-
administration.org/article/317/An_introduction_to_bash_completion_part_2

dHutchings committed:
Impliment Tab-completion of device names for ./flash.sh

Only works on linux. However, since the setup for compilation of hibike
also only works on linux, I think this is fine.

Done based on this documenation: https://debian-
administration.org/article/317/An_introduction_to_bash_completion_part_2

nikitakit commented on hibike/utils/flash:
```diff
@@ -0,0 +1,26 @@
+    prev="${COMP_WORDS[COMP_CWORD-1]}"
+    opts="$(./utils/device_names.sh)"
+ 
+	COMPREPLY=( $(compgen -W "${opts}" -- ${cur}))
```
Can you fix the inconsistent indentation?

dHutchings replied:
Done

nikitakit commented on hibike/utils/flash:
```diff
@@ -0,0 +1,26 @@
+	COMPREPLY=( $(compgen -W "${opts}" -- ${cur}))
+	return 0
+}
+complete -F _flash.sh flash.sh
```
newline at eof

dHutchings replied:
done

nikitakit commented on hibike/utils/device_names.sh:
```diff
@@ -0,0 +1,10 @@
+#it is meant to be called from the "hibike" folder, since that is where flash.sh is located.
+
+
+ls devices/
```
Could you add a newline at the end of the file?

(See how there's a red icon when github displays this line? That means there's no newline at eof. It's standard practice to end a file in a newline, because of how lack of a newline interacts with diffing tools)

dHutchings replied:
Done

nikitakit reviewed (changes requested):
I have a few style comments.

There's probably a way to make this more robust and not call `device_names.sh` from random locations (as well as not complete when device names don't exist), but for an internal tool that's not really necessary.

dHutchings committed:
Modified flash.sh

Now, attempts to run device_names.sh only if the user's terminal is
located in PieCentral/hibike

Style Fixes

dHutchings committed:
Added newline at EOF.  Also added helpful comments

Now the install script reminds the user to restart their terminal, to
get tab-complete.

dHutchings commented:
@nikitakit 

Fixed your style comments.  Also, implemented extra logic so that it won't erroneously attempt to run the device_names script unless the user's terminal is located in the correct filepath.
